src/routes/Apps/App/Settings.tsx

```diff
@@ -229,10 +229,16 @@ class Settings extends React.Component<Props, ComponentState> {
     }
 
     onGetNameErrorMessage(value: string): string {
+        const MAX_NAME_LENGTH = 30
+
         if (value.length === 0) {
             return Util.formatMessageId(this.props.intl, FM.SETTINGS_FIELDERROR_REQUIREDVALUE)
         }
 
+        if (value.length > MAX_NAME_LENGTH) {
+            return Util.formatMessageId(this.props.intl, FM.APPCREATOR_FIELDERROR_TOOLONG)
```

Comment:
Technically this is wrong since it's an `APPCREATOR` field, but in order to fix I'd have to change both settings and app creator to use generic / shareable names.